classlib/src/main/java/org/teavm/classlib/java/lang/TString.java

```diff
@@ -80,30 +81,34 @@ static TString fromArray(char[] characters) {
     private native void takeCharArray(char[] characters);
 
     public TString(byte[] bytes, int offset, int length, TString charsetName) throws TUnsupportedEncodingException {
-        this(bytes, offset, length, TCharset.forName(charsetName.toString()));
+        Objects.checkFromIndexSize(offset, length, bytes.length);
+        initWithBytes(bytes, offset, length, TCharset.forName(charsetName.toString()));
     }
 
     public TString(byte[] bytes, int offset, int length, TCharset charset) {
+        Objects.checkFromIndexSize(offset, length, bytes.length);
         initWithBytes(bytes, offset, length, charset);
     }
 
     public TString(byte[] bytes, int offset, int length) {
+        Objects.checkFromIndexSize(offset, length, bytes.length);
         initWithBytes(bytes, offset, length, TUTF8Charset.INSTANCE);
     }
 
     public TString(byte[] bytes) {
-        this(bytes, 0, bytes.length);
+        initWithBytes(bytes, 0, bytes.length, TUTF8Charset.INSTANCE);
     }
 
     public TString(byte[] bytes, TString charsetName) throws TUnsupportedEncodingException {
-        this(bytes, 0, bytes.length, charsetName);
+        initWithBytes(bytes, 0, bytes.length, TCharset.forName(charsetName.toString()));
     }
 
     public TString(byte[] bytes, TCharset charset) {
-        this(bytes, 0, bytes.length, charset);
+        initWithBytes(bytes, 0, bytes.length, charset);
     }
 
     public TString(int[] codePoints, int offset, int count) {
+        Objects.checkFromIndexSize(offset, count, codePoints.length);
```

Comment:
The code below reads from the array, which in strict mode will throw AIIOBE. So is this check really necessary?